UngradedModules:   AssertionError: is_welldefined(M3_to_M1)

Error during github tests for master with julia nightly on macos: https://github.com/oscar-system/Oscar.jl/actions/runs/6362233902/job/17277140389#step:6:1077
```
tensoring morphisms: Error During Test at /Users/runner/work/Oscar.jl/Oscar.jl/test/Modules/UngradedModules.jl:663
  Got exception outside of a @test
  AssertionError: is_welldefined(M3_to_M1)
  Stacktrace:
    [1] macro expansion
      @ ~/work/Oscar.jl/Oscar.jl/test/Modules/UngradedModules.jl:696 [inlined]
    [2] macro expansion
      @ ~/hostedtoolcache/julia/nightly/x64/share/julia/stdlib/v1.11/Test/src/Test.jl:1598 [inlined]
    [3] top-level scope
      @ ~/work/Oscar.jl/Oscar.jl/test/Modules/UngradedModules.jl:664
    [4] include
      @ ./Base.jl:498 [inlined]
    [5] macro expansion
      @ ./timing.jl:506 [inlined]
    [6] include(str::String)
      @ Main ~/work/Oscar.jl/Oscar.jl/test/runtests.jl:79
    [7] top-level scope
      @ ~/work/Oscar.jl/Oscar.jl/test/Modules/runtests.jl:4
    [8] include
      @ ./Base.jl:498 [inlined]
    [9] macro expansion
      @ ./timing.jl:506 [inlined]
   [10] include(str::String)
      @ Main ~/work/Oscar.jl/Oscar.jl/test/runtests.jl:79
   [11] (::Base.var"#1054#1059"{typeof(include)})(r::Base.RefValue{Any}, args::Tuple{String})
      @ Base ./asyncmap.jl:94
   [12] (::Base.var"#1070#1071"{Base.var"#1054#1059"{typeof(include)}, Channel{Any}, Nothing})()
      @ Base ./asyncmap.jl:228
Test Summary:       | Pass  Error  Total  Time
tensoring morphisms |    3      1      4  1.8s
ERROR: LoadError: LoadError: Some tests did not pass: 3 passed, 0 failed, 1 errored, 0 broken.
```

Ok, I needed to do the full test-file instead of just trying that testset. With some debug output on julia-nightly from today:
```julia
Test Summary: | Pass  Total  Time
Hom module    |  152    152  2.5s
A1 = [6*x^4*y^2*z+6*x*y^4*z^2+11*x*y*z^4 14*x*y^2+12*y^2*z^3+5*y; 15*x^2*y^2*z^4+12*x*y^2+13*x*z^2 15*x^4*y^2*z^2; 2*x*y*z^3+9*x 15*x^4*y^3*z^3+14*x^3*y^4*z+6*x^3*y^2]
B1 = [11*x*y^2*z^2 12*z^2]
A3 = [0 2*x^2*y*z; 0 5*y*z]
tmp = [1 2*x^2*y^2*z^2 y^2*z^2+z^2; 2*y^2*z^2 2*x*y*z^2+2 z]
M3_to_M1 = Map with following data
Domain:
=======
Submodule with 2 generators
1 -> 2*x^2*y*z*e[2]
2 -> 5*y*z*e[2]
represented as subquotient with no relations.
Codomain:
=========
Subquotient of Submodule with 3 generators
1 -> (6*x^4*y^2*z + 6*x*y^4*z^2 + 11*x*y*z^4)*e[1] + (14*x*y^2 + 12*y^2*z^3 + 5*y)*e[2]
2 -> (15*x^2*y^2*z^4 + 12*x*y^2 + 13*x*z^2)*e[1] + 15*x^4*y^2*z^2*e[2]
3 -> (2*x*y*z^3 + 9*x)*e[1] + (15*x^4*y^3*z^3 + 14*x^3*y^4*z + 6*x^3*y^2)*e[2]
by Submodule with 1 generator
1 -> 11*x*y^2*z^2*e[1] + 12*z^2*e[2]

tensoring morphisms: Error During Test at /home/lorenz/software/polymake/julia/Oscar.jl/test/Modules/UngradedModules.jl:663
  Got exception outside of a @test
  AssertionError: is_welldefined(M3_to_M1)
  Stacktrace:
    [1] macro expansion
      @ ~/software/polymake/julia/Oscar.jl/test/Modules/UngradedModules.jl:701 [inlined]
    [2] macro expansion
      @ ~/software/polymake/julia/julia/julia-nightly/share/julia/stdlib/v1.11/Test/src/Test.jl:1598 [inlined]
    [3] top-level scope
      @ ~/software/polymake/julia/Oscar.jl/test/Modules/UngradedModules.jl:664
    [4] include
      @ ./Base.jl:497 [inlined]
    [5] test_module(file::String; new::Bool)
      @ Oscar ~/software/polymake/julia/Oscar.jl/src/utils/tests.jl:25
    [6] top-level scope
      @ REPL[15]:1
    [7] eval
      @ ./boot.jl:396 [inlined]
    [8] eval_user_input(ast::Any, backend::REPL.REPLBackend, mod::Module)
      @ REPL ~/software/polymake/julia/julia/julia-nightly/share/julia/stdlib/v1.11/REPL/src/REPL.jl:154
    [9] repl_backend_loop(backend::REPL.REPLBackend, get_module::Function)
      @ REPL ~/software/polymake/julia/julia/julia-nightly/share/julia/stdlib/v1.11/REPL/src/REPL.jl:250
   [10] start_repl_backend(backend::REPL.REPLBackend, consumer::Any; get_module::Function)
      @ REPL ~/software/polymake/julia/julia/julia-nightly/share/julia/stdlib/v1.11/REPL/src/REPL.jl:235
   [11] run_repl(repl::REPL.AbstractREPL, consumer::Any; backend_on_current_task::Bool, backend::Any)
      @ REPL ~/software/polymake/julia/julia/julia-nightly/share/julia/stdlib/v1.11/REPL/src/REPL.jl:391
   [12] run_repl(repl::REPL.AbstractREPL, consumer::Any)
      @ REPL ~/software/polymake/julia/julia/julia-nightly/share/julia/stdlib/v1.11/REPL/src/REPL.jl:377
   [13] (::Base.var"#1044#1046"{Bool, Symbol, Bool})(REPL::Module)
      @ Base ./client.jl:449
   [14] #invokelatest#2
      @ Base ./essentials.jl:899 [inlined]
   [15] invokelatest
      @ Base ./essentials.jl:896 [inlined]
   [16] run_main_repl(interactive::Bool, quiet::Bool, banner::Symbol, history_file::Bool, color_set::Bool)
      @ Base ./client.jl:432
   [17] exec_options(opts::Base.JLOptions)
      @ Base ./client.jl:332
   [18] _start()
      @ Base ./client.jl:569
Test Summary:       | Pass  Error  Total  Time
tensoring morphisms |    3      1      4  1.2s
ERROR: LoadError: Some tests did not pass: 3 passed, 0 failed, 1 errored, 0 broken.
in expression starting at /home/lorenz/software/polymake/julia/Oscar.jl/test/Modules/UngradedModules.jl:663
```

And to reproduce this error with different julia versions:
```julia
julia> R, (x,y,z) = polynomial_ring(QQ, ["x", "y", "z"]);

julia> F2 = FreeMod(R,2);

julia> F3 = FreeMod(R,3);

julia> A1 = matrix([6*x^4*y^2*z+6*x*y^4*z^2+11*x*y*z^4 14*x*y^2+12*y^2*z^3+5*y; 15*x^2*y^2*z^4+12*x*y^2+13*x*z^2 15*x^4*y^2*z^2; 2*x*y*z^3+9*x 15*x^4*y^3*z^3+14*x^3*y^4*z+6*x^3*y^2]);

julia> B1 = matrix([11*x*y^2*z^2 12*z^2]);

julia> A3 = matrix([zero(R) 2*x^2*y*z; zero(R) 5*y*z]);

julia> M1 = SubquoModule(F2,A1,B1);

julia> M3 = SubquoModule(Oscar.SubModuleOfFreeModule(F2,A3));

julia> tmp = matrix([one(R) 2*x^2*y^2*z^2 y^2*z^2+z^2; 2*y^2*z^2 2*x*y*z^2+2 z]);

julia> M3_to_M1 = SubQuoHom(M3,M1, tmp)
Map with following data
Domain:
=======
Submodule with 2 generators
1 -> 2*x^2*y*z*e[2]
2 -> 5*y*z*e[2]
represented as subquotient with no relations.
Codomain:
=========
Subquotient of Submodule with 3 generators
1 -> (6*x^4*y^2*z + 6*x*y^4*z^2 + 11*x*y*z^4)*e[1] + (14*x*y^2 + 12*y^2*z^3 + 5*y)*e[2]
2 -> (15*x^2*y^2*z^4 + 12*x*y^2 + 13*x*z^2)*e[1] + 15*x^4*y^2*z^2*e[2]
3 -> (2*x*y*z^3 + 9*x)*e[1] + (15*x^4*y^3*z^3 + 14*x^3*y^4*z + 6*x^3*y^2)*e[2]
by Submodule with 1 generator
1 -> 11*x*y^2*z^2*e[1] + 12*z^2*e[2]


julia> is_welldefined(M3_to_M1)
false
```

--- a/test/Modules/UngradedModules.jl
+++ b/test/Modules/UngradedModules.jl
@@ -693,7 +693,7 @@ end
     N,pure_N = tensor_product(M3,F4, task=:map)
 
     M3_to_M1 = SubQuoHom(M3,M1, matrix([randpoly(R,0:2,2,2) for i=1:ngens(M3), j=1:ngens(M1)]))
-    @assert is_welldefined(M3_to_M1)
+    is_welldefined(M3_to_M1) || continue
     F4_to_M2 = FreeModuleHom(F4,M2, matrix([randpoly(R,0:2,2,2) for i=1:ngens(F4), j=1:ngens(M2)]))
 
     phi = hom_tensor(N,M,[M3_to_M1,F4_to_M2])

